### PR TITLE
Adding algolia search

### PR DESCRIPTION
**Pull request by signorecello**
Adding full-site search using Algolia

### CODE REVIEW

**Comment by critesjosh** [comment]
Do we want all of the optional parameters included? I don't think the main docs site has any fwiw.

**Comment by signorecello** [comment]
Just copied/pasted the JSON from the guide, search looks great so I left them... But lmk if you prefer to remove them

File: docusaurus.config.js [hunk 1]
```diff
@@ -130,6 +119,35 @@ const config = {
...
+        // Public API key: it is safe to commit it
+        apiKey: '8cdbbd8b0b98caab7765766267b1cb23',
+
+        indexName: 'noir-lang',
+
+        // Optional: see doc section below
+        contextualSearch: true,
+
+        // Optional: Specify domains where the navigation should occur through window.location instead on history.push. Useful when our Algolia config crawls multiple documentation sites and we want to navigate with window.location.href to them.
+        externalUrlRegex: 'external\\.com|domain\\.com',
```
Comment (phated) [inline docusaurus.config.js:135]: I think this was supposed to be `noir-lang.org`, right?

Comment (phated) [reply docusaurus.config.js:135]: Oh, this is actually supposed to be external sites that we link out to that algolia might crawl. So probably just fine to remove it

**Pull request merged**